lib/facility.rb

```diff
@@ -1,14 +1,72 @@
+require './lib/vehicle'
+require './lib/registrant'
+require './lib/facility_factory'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :collected_fees,
+              :registered_vehicles,
+              :daily_hours,
+              :holiday_closed
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(input)
+    @name = input[:name]
+    @address = input[:address]
+    @phone = input[:phone]
     @services = []
+    @collected_fees = 0
+    @plate_type = {}
+    @registered_vehicles = []
+    @daily_hours = input[:daily_hours]
+    @holiday_closed = input[:holiday_closed]
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def collect_fees(vehicle)
+    if vehicle.plate_type == :ev
+      @collected_fees  += 200
+    elsif vehicle.plate_type == :antique
+      @collected_fees += 25
+    else
+      @collected_fees += 100
+    end
+  end
+
+  def register_vehicle(vehicle)
```

Comment:
So this method is actually supposed to be doing a bit more work. It should be adding the vehicle to the registered_vehicles array like you have in line 43, but it should be also setting the registration date of the vehicle and setting the plate_type of the vehicle.